src/Components/Pages/Data/Calculator.js

```diff
@@ -38,7 +40,9 @@ function toNumericInput(inputArray, separator){
 function transformInputs(inputArray, separator){
     const numericInputs = toNumericInput(inputArray, separator)
     if(numericInputs.some(ni => isNaN(ni))){
-        throw EvalError("Some of the inputs are non-numeric.")
+        throw new NonNumericError()
+    } else if (numericInputs.some(ni => ni < 0)){
```

Comment:
Can it be `ni <= 0` ?

Reply:
0 is not an error - in fact it always appears in the input when trying to type numbers smaller than 1. In other words you always have to have a standalone 0 before you have a 0.5. If it was an error you wouldn't be able to input those values

Reply:
One thing we didn't think of earlier would be updating the results only when the new line would be added - this way standalone 0 could be treated as an error. That change includes somewhat major refactoring so i wouldn't want to do it in this PR tho.